libp2p/protocols/pubsub/gossipsub.nim

```diff
@@ -170,13 +170,19 @@ proc dropFanoutPeers(g: GossipSub) {.async.} =
     if Moment.now > val:
       dropping.add(topic)
       g.fanout.del(topic)
+      trace "dropping fanout topic", topic
 
   for topic in dropping:
     g.lastFanoutPubSub.del(topic)
 
     libp2p_gossipsub_peers_per_topic_fanout
       .set(g.fanout.getOrDefault(topic).len.int64, labelValues = [topic])
 
+proc shuffle[T](collection: var seq[T]) = 
```

Comment:
Why do a re-shuffle on every drop? Doesn't seem necessary and doesn't seem part of the spec either.

Reply:
well before it was `sample`, now it's `shuffle`, same thing just batched, spec says random